### ISSUE TITLE
[Converters] Automate migration from Propritery vendors

### ISSUE THREAD

**Issue by nrshrivatsan** [envoyproxy/envoy-tools#7]
# Why

- Potential users of Envoy need to migrate legacy systems from Proprietary vendors
- Envoy's Control Plane APIs are awesome, but seamless migration from legacy to Software Layer-7 Proxies is expensive and cost-inhibiting
- Open automation tools to migrate from Proprietary DSLs into Envoy config syntax could save a lot of time for Envoy Users 

# What
- Proprietary vendors have open specs
- Create a syntax tree parser to migrate from a vendor's DSL into Envoy config
- Provide automated *validation* scripts to validate the conversion 

## References
- [Netscaler syntax](https://www.citrix.com/content/dam/citrix/en_us/citrix-developer/documents/Netscaler/irules/rewrite-http-redirect-response.pdf)
- [F5](https://devcentral.f5.com/articles/irules-101-10-regular-expressions)
- [Zevenet-API](https://www.zevenet.com/zapidocv3.2/#modify-a-service)

**Comment by htuch** [comment]
@nrshrivatsan agreed that this would be useful. Do you have plans to contribute here or should we mark this as "help wanted"?
